tests/src/python/test_qgsdelimitedtextprovider.py

```diff
@@ -188,7 +239,10 @@ def layerData(self, layer, request={}, offset=0):
                 for field in f.fields():
                     fields.append(str(field.name()))
                     fieldTypes.append(str(field.typeName()))
-            fielddata = dict((name, unicode(f[name])) for name in fields)
+            if sys.version_info.major == 2:
+                fielddata = dict((name, unicode(f[name])) for name in fields)
+            else:
+                fielddata = dict((name, str(f[name])) for name in fields)
```

Comment:
Does

    name for name in fields

or

   repr(name) for name in fields

not work cross-version?